packages/components/src/Actions/ActionFile/ActionFile.component.js

```diff
@@ -0,0 +1,137 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { OverlayTrigger } from 'react-bootstrap';
+import classNames from 'classnames';
+import uuid from 'uuid';
+
+import TooltipTrigger from '../../TooltipTrigger';
+import CircularProgress from '../../CircularProgress';
+
+import Icon from '../../Icon';
+import theme from './ActionFile.scss';
+
+const LEFT = 'left';
+const RIGHT = 'right';
+
+function getIcon({ icon, iconTransform, inProgress }) {
+	if (inProgress) {
+		return <CircularProgress size="small" key="icon" />;
+	}
+
+	if (icon) {
+		return <Icon name={icon} transform={iconTransform} key="icon" />;
+	}
+
+	return null;
+}
+getIcon.propTypes = {
+	icon: PropTypes.string,
+	iconTransform: PropTypes.string,
+	inProgress: PropTypes.bool,
+};
+
+function getLabel({ hideLabel, label }) {
+	if (hideLabel) {
+		return null;
+	}
+	return <span key="label">{label}</span>;
+}
+
+getLabel.propTypes = {
+	label: PropTypes.string,
+	hideLabel: PropTypes.bool,
+};
+
+function adjustContentPlacement(icon, label, iconPosition) {
+	if (iconPosition === RIGHT) {
+		return [label, icon];
+	}
+	return [icon, label];
+}
+
+function getContent(props) {
+	return adjustContentPlacement(getIcon(props), getLabel(props), props.iconPosition);
+}
+
+function noOp() {}
+
+function ActionFile(props) {
+	const {
+		id,
+		bsStyle,
+		inProgress,
+		disabled,
+		hideLabel,
+		label,
+		name,
+		onChange = noOp,
+		tooltipPlacement,
+		tooltip,
+		tooltipLabel,
+		available,
+		...rest
+	} = props;
+
+	if (!available) {
+		return null;
+	}
+	const localId = id || uuid.v4();
+	const buttonContent = getContent(props);
+	const labelClasses = classNames('btn', theme['btn-file'], (disabled || inProgress) && 'disabled');
+	const localOnChange = event => {
+		event.preventDefault();
+		if (onChange) {
+			onChange(event, event.target.files[0]);
+		}
+	};
+	const btn = (
+		<span>
+			<input
+				onChange={localOnChange}
+				type="file"
+				name={name}
+				id={localId}
+				disabled={inProgress || disabled}
+				className={theme['action-file-label']}
+			/>
+			<label htmlFor={localId} className={labelClasses}>
+				{buttonContent}
+			</label>
+		</span>
+	);
```

Comment:
I'm not sure that constant if useful if still defined in the same function. :thinking:

Reply:
because this defined const can be used directly in the function to be returned wrapped into another component or returned directly